### ISSUE TITLE
I want to fetch only the last 10 comments of a video. How can I do exactly this without downloading all the comments?

### ISSUE THREAD

**Issue by Roxy45l** [yt-dlp/yt-dlp#10434]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I think the max-comments parameter is not suitable for this situation. Because I need to check whether a new person has been added after the last comment according to the comments id.
```py
def collect_comments(video_url, output_dir="comments"):
    if not os.path.exists(output_dir):
        os.makedirs(output_dir)

    max_comments = 100

    ydl_opts = {
        'verbose': True,
        'getcomments': True,
        'extractor_args': {
            'youtube': {
                'max_comments': [str(max_comments)],
            }
        }
    }
    try:
        with YoutubeDL(ydl_opts) as ydl:
            result = ydl.extract_info(video_url, download=False)
            if 'comments' in result and result['comments']:
                comments = result['comments']
                video_id = result['id']


                # file_path = os.path.join(output_dir, f"{video_id}_comments.json")
                #
                # with open(file_path, 'w', encoding='utf-8') as json_file:
                #     json.dump(comments, json_file, ensure_ascii=False, indent=4)

                return comments

    except Exception as e:
        print(f"getting err: {e}")
        if 'Private video' in str(e):
            print("removed")
```

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by bashonly** [comment]
Can you rephrase the question, or elaborate on your goal? I don't quite understand what you are trying to achieve, and I don't understand why the extractor-arg is not sufficient.


---

- `max_comments`: Limit the amount of comments to gather. Comma-separated list of integers representing `max-comments,max-parents,max-replies,max-replies-per-thread`. Default is `all,all,all,all`

    - E.g. `all,all,1000,10` will get a maximum of 1000 replies total, with up to 10 replies per thread. `1000,all,100` will get a maximum of 1000 comments, with a maximum of 100 replies total

**Comment by Roxy45l** [comment]
@bashonly  I stated that I wanted to get the last 10 comments of a video. For example, if there are 100 comments on a video, I want to get the 10 most recent comments. I hope I could explain myself

**Comment by bashonly** [comment]
Try this
```py
ydl_opts = {
    'extractor_args': {
        'youtube': {
            'max_comments': ['10'],
        },
    },
}
```

**Comment by Roxy45l** [comment]
@bashonly Dude I tried this and it's not working. downloading all comments

**Comment by seproDev** [comment]
You will have to provide a bit more info than does not work. I just tried it, and it is working fine for me. Can you provide a verbose log?

**Comment by PexMor** [comment]
@Roxy45l Try to add the other params it seems to work as expected then:

```python
       "extractor_args": {
            "youtube": {
                "max_comments": [
                    str(max_comments),
                    str(max_parents),
                    str(max_replies),
                    str(max_replies_per_thread),
                ],
            }
        },
```
